packages/esm-service-queues-app/src/add-provider-queue-room/add-provider-queue-room.component.tsx

```diff
@@ -33,200 +35,261 @@ import {
+                control={control}
+                name="queueProviderMapUuid"
+                render={({ field }) => (
+                  <Dropdown
+                    {...field}
+                    aria-label={t('queueService', 'Queue service')}
+                    id="service"
+                    initialSelectedItem={{
+                      uuid: currentService?.serviceUuid,
+                      display: currentService?.serviceDisplay,
+                    }}
+                    items={services ?? []}
+                    itemToString={(item) => (item ? item.display : '')}
+                    label="Service"
+                    onChange={(e) => {
+                      field.onChange(e.selectedItem?.uuid);
+                      handleServiceChange(e);
+                    }}
+                    titleText={t('queueService', 'Queue service')}
+                  />
+                )}
+              />
+            </section>
+
+            <section>
+              <Controller
+                control={control}
+                name="queueRoomUuid"
+                render={({ field }) => (
+                  <Select
+                    {...field}
+                    disabled={errorFetchingQueueRooms}
+                    id="room"
+                    invalidText={errors.queueRoomUuid?.message}
+                    invalid={!!errors.queueRoomUuid}
+                    labelText={t('queueRoom', 'Queue room')}
+                    onChange={(event) => {
+                      field.onChange(event.target.value);
+                    }}>
+                    <SelectItem text={t('queueRoom', 'Queue room')} value="" />
+                    {rooms?.length > 0 &&
+                      rooms?.map((room) => <SelectItem key={room.uuid} text={room.display} value={room.uuid} />)}
+                  </Select>
+                )}
+              />
+              {errorFetchingQueueRooms && (
+                <InlineNotification
+                  className={styles.errorNotification}
+                  kind="error"
+                  onClick={() => {}}
+                  subtitle={''}
+                  title={t('errorFetchingQueueRooms', 'Error fetching queue rooms')}
+                />
+              )}
+            </section>
+
             <section>
-              <InlineNotification
-                style={{ margin: '0', minWidth: '100%' }}
-                kind="error"
-                lowContrast={true}
-                title={t('missingQueueRoom', 'Please select a queue room')}
+              <Controller
+                control={control}
+                name="currentIsPermanentProviderQueueRoom"
+                render={({ field }) => (
+                  <Checkbox
+                    checked={field.value}
+                    className={styles.checkbox}
+                    id="retainLocation"
+                    labelText={t('retainLocation', 'Retain location')}
+                    onChange={handleRetainLocation}
```

Comment:
```suggestion
                    onChange={(event, { checked }) => {
                      field.onChange(checked);
                      handleRetainLocation(event);
                    }}
```

We could do something like this, and then simplify the handler function to: 

```tsx
const handleRetainLocation = useCallback((isChecked: boolean) => {
  localStorage.setItem('isPermanentProviderQueueRoom', String(isChecked));
  updateIsPermanentProviderQueueRoom(isChecked);
}, []);
```